File: 2015/FASE 2/C++/q-erradas-c++/catador.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int N, M;
    cin >> N >> M;

    vector<int> baldes(N);
    for (int i = 0; i < N; ++i) {
        cin >> baldes[i];
    }

    vector<int> operacoes(M);
    for (int i = 0; i < M; ++i) {
        cin >> operacoes[i];
    }

    int totalConchas = 0;
    for (int i = 0; i < M; ++i) {
        int indice = operacoes[i] - 1;
        int C = baldes[indice];
        totalConchas += C;
        int inicio = max(0, indice - C);
        int fim = min(N - 1, indice + C);
        for (int j = inicio; j <= fim; ++j) {
            baldes[j] = max(0, baldes[j] - 1);
        }
    }

    cout << totalConchas << endl;

    return 0;
}

File: 2015/FASE 2/PYTHON/q-erradas-py/catador.py
def catador_de_conchas(N, M, quantidades, operacoes):
    total_conchas = sum(quantidades)  # Total de conchas inicialmente

    for indice in operacoes:
        conchas_no_baldes = quantidades[indice - 1]  # Quantidade de conchas no balde atual

        # Remover uma concha de cada balde dentro do alcance especificado
        for j in range(max(1, indice - conchas_no_baldes), min(N + 1, indice + conchas_no_baldes + 1)):
            quantidades[j - 1] -= 1

    # Calcular o total de conchas restantes
    total_conchas_restantes = sum(quantidades)

    return total_conchas_restantes

# Leitura da entrada
N, M = map(int, input().split())  # Número de baldes e operações
quantidades = list(map(int, input().split()))  # Quantidades de conchas em cada balde
operacoes = list(map(int, input().split()))  # Índices das operações

# Chamar a função e imprimir o resultado
print(catador_de_conchas(N, M, quantidades, operacoes))
